### ISSUE TITLE
Strange issue with repl in scala 2.11.5

### ISSUE THREAD

**Issue by tylerprete** [scala/scala#4339]
I don't see this in 2.10.4 or 2.11.2, but see it consistently in 2.11.5.

```
scala> val q = List((1,2))
q: List[(Int, Int)] = List((1,2))

scala> q.groupBy(_._1)
java.lang.AssertionError: assertion failed: List(value _1$mcI$sp, value _1$mcI$sp, value _1$mcI$sp)
	at scala.reflect.internal.Symbols$Symbol.suchThat(Symbols.scala:1916)
	at scala.tools.nsc.transform.SpecializeTypes$$anon$2.matchingSymbolInPrefix$1(SpecializeTypes.scala:1460)

...

	at scala.tools.nsc.MainGenericRunner.run$1(MainGenericRunner.scala:87)
	at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:98)
	at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:103)
	at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)

That entry seems to have slain the compiler.  Shall I replay
your session? I can re-run each line except the last one.
[y/n]
```

Also, it seems to be related to assignment, because it's not triggered when combined as a single expression.

```
scala> val q = List((1,2)).groupBy(_._1)
q: scala.collection.immutable.Map[Int,List[(Int, Int)]] = Map(1 -> List((1,2)))
```

**Comment by retronym** [comment]
Thanks for reporting.

This [bug](https://issues.scala-lang.org/browse/SI-9089) will be fixed in the forthcoming release, 2.11.6.

**Comment by som-snytt** [comment]
:clock11:
